client_apis/python/src/tests/integration/user_test.py

```diff
@@ -0,0 +1,252 @@
+#
+# CARBON BLACK API TESTS - user
+# Copyright, Bit9, Inc 2015
+#
+
+""" These tests require CarbonBlack Enterprise Server to be installed.
+    You can run this script by passing the server URL and user API token
+    as parameters.
+"""
+
+import unittest
+import sys
+import os
+import requests
+import uuid
+
+if __name__ == '__main__':
+    sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), "../../")))
+
+from cbapi.cbapi import CbApi
+
+cb = None
+
+class CbApiUserTest(unittest.TestCase):
+    class TestDataGen:
+        @staticmethod
+        def gen_uid_hex():
+            return uuid.uuid4().hex
+
+        @classmethod
+        def gen_user(cls):
+            uid_hex = cls.gen_uid_hex()
+
+            user = {
+                'username': 'testuser ' + uid_hex,
+                'first_name': 'IntegrationUser_' + uid_hex,
+                'last_name': 'TestUser',
+                'email': 'integration.testuser ' + uid_hex + '@example.com',
+                'password': 'p@ssw0rd',
+                'global_admin': False,
+                'teams': []
+            }
+            return user
+
+        @classmethod
+        def gen_global_admin(cls):
+            user = cls.gen_user()
+            user['global_admin'] = True
+
+            return user
+
+    ## List user tests
+
+    def test_list_users(self):
+        users = cb.user_enum()
+        self.assertIsNotNone(users)
+        self.assertGreater(len(users), 0)
+
+        for user in users:
+            self._verify_retrieved_user(user)
+
+    ## Get user tests
+
+    def test_get_current_user(self):
+        cur_user = cb.user_info(None)
+        self.assertIsNotNone(cur_user)
+
+        self._verify_retrieved_user(cur_user)
+
+    ## Create user tests
+
+    def test_create_user_no_username(self):
+        user_data = self.TestDataGen.gen_user()
+
+        user_add_params = self._convert_user_data_to_user_add_params(user_data)
+        user_add_params['username'] = None
+
+        # should not be able to create user with no username
+        with self.assertRaises(requests.HTTPError) as cm:
```

Comment:
I've been pondering how important it is to know which HTTP status codes are returned in these error cases. The watchlist tests are incosistent (mostly because I didn't know what to expect half the time). Thoughts?